wps_shared/wps_shared/db/database.py

```diff
@@ -93,6 +97,19 @@ async def get_async_write_session_scope() -> AsyncGenerator[AsyncSession, None]:
     finally:
         await session.close()
 
+@asynccontextmanager
+async def get_async_write_session_scope_no_expire() -> AsyncGenerator[AsyncSession, None]:
```

Comment:
When a new fire watch is posted to the back end it gets saved and the updated record ( including id, created by, created time, updated by and updated time) gets returned back to the front-end in the response. The best way to get the new id is to access the ORM object after calling `session.commit()` but it is necessary to set `expire_on_commit = False` on the session in order to be able to access properties of the ORM object after the commit which is the reason for this alternate session generator.

Reply:
Did you check out `returning`? 

https://docs.sqlalchemy.org/en/13/core/dml.html#sqlalchemy.sql.expression.Insert.returning

Reply:
I looked into it but you are limited to working with `Table` objects to perform an `insert` of values. It would work great in a migration where we declare `Table` objects but doesn't work with ORM objects/classes.

Reply:
We use it in HFI Calculator like this:

```
    stmt = (
        insert(HFIRequest).values(fire_centre_id=hfi_result_request.selected_fire_center_id,
                                  prep_start_day=hfi_result_request.date_range.start_date,
                                  prep_end_day=hfi_result_request.date_range.end_date,
                                  create_timestamp=get_utc_now(),
                                  create_user=username,
                                  request=hfi_result_request.json()).returning(HFIRequest.id)
    )
    res = session.execute(stmt).fetchone()
    ```

Reply:
Thanks for pointing this out! I've adopted this approach and removed that extra async session generator.